src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+            coords=coords,
+        )
+
+    (target_var,) = target_data.data_vars
+    (independent_var,) = independent_data.data_vars
+    # pred_target_var = target_var
+    pred_independent_var = independent_var
+
+    if prediction_data is None:
+        prediction_data = process_group_variables_coords(
+            dt, group=prediction_group, var_names=target_var, filter_vars=filter_vars, coords=coords
+        )
+    elif isinstance(prediction_data, str):
+        # pred_target_var = prediction_data
+        prediction_data = process_group_variables_coords(
+            dt,
+            group=prediction_group,
+            var_names=prediction_data,
+            filter_vars=filter_vars,
+            coords=coords,
+        )
+
+    if ci_mean_independent_data is None:
+        if prediction_group == "predictions":
+            ci_mean_independent_data = process_group_variables_coords(
+                dt,
+                group="predictions_constant_data",
+                var_names=independent_var,
+                filter_vars=filter_vars,
+                coords=coords,
+            )
+        else:
+            ci_mean_independent_data = independent_data
+    elif isinstance(ci_mean_independent_data, str):
+        if prediction_group == "predictions":
+            pred_independent_var = ci_mean_independent_data
+            ci_mean_independent_data = process_group_variables_coords(
+                dt,
+                group="predictions_constant_data",
+                var_names=ci_mean_independent_data,
+                filter_vars=filter_vars,
+                coords=coords,
+            )
+        else:
+            ci_mean_independent_data = independent_data
+
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+    bg_color = plot_bknd.get_background_color()
+    _, contrast_gray_color = get_contrast_colors(bg_color=bg_color, gray_flag=True)
+    if plot_collection is None:
+        pc_kwargs.setdefault("cols", None)
+        pc_kwargs["figure_kwargs"] = pc_kwargs.get("figure_kwargs", {}).copy()
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        pc_kwargs = set_wrap_layout(pc_kwargs, plot_bknd, prediction_data)
+        plot_collection = PlotCollection.wrap(
+            prediction_data,
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    if aes_by_visuals is None:
+        aes_by_visuals = {}
+    else:
+        aes_by_visuals = aes_by_visuals.copy()
+    aes_by_visuals.setdefault("line", plot_collection.aes_set)
+
+    # calculations for credible interval
+    if ci_kind == "eti":
+        ci = prediction_data.azstats.eti(prob=ci_prob, **stats.get("credible_interval", {}))
+    elif ci_kind == "hdi":
+        ci = prediction_data.azstats.hdi(prob=ci_prob, **stats.get("credible_interval", {}))
+    ci_upper = ci.y.sel(ci_bound="upper").values
+    ci_lower = ci.y.sel(ci_bound="lower").values
+
+    mean_data = prediction_data.mean(dim=["chain", "draw"])
```

Comment:
We should offer an option to compute the mean and median. 
Should we also offer an option to "smooth" the results?